### DISCUSSION TITLE
visualizing sgRNA-response pairs with Nextflow output

### DISCUSSION THREAD

**Discussion by nlthanhthanh** [Katsevich-Lab/sceptre#148] [Q&A] [answered]
Hello -- big fan of sceptre here!

I successfully ran Nextflow Sceptre through to the final steps of association analysis. I was wondering if there is any way to produce sgRNA-response pairs plots similarly to plot_response_grna_target_pair function using Nextflow outputs (which include gRNA assignment matrix and .RDS files with summary statistics). Many thanks!

Thanh Thanh

**Comment by timothy-barry** [comment, marked as answer]
Hi, good to hear!

Would it be feasible for you to run the gRNA assignment and QC steps within the R console? If so, my recommendation would be to run the pipeline out-of-core up through `run_qc()` within the R console (according to [these instructions](https://timothy-barry.github.io/sceptre-book/pipeline-args.html#sec-analyze_odm_within_r)). Then, you should be able to call `plot_response_grna_target_pair()` on the `sceptre_object`. Note that the p-value will not be shown; you will have to superimpose this yourself. I think this should work, but please let me know if you hit any snags.

**Comment by nlthanhthanh** [comment]
Thank you for your response! My dataset is big so it wasn't feasible to run out-of-core. However, I dug through the Nextflow cache files and were able to locate the sceptre_object at the run_qc step!

Which unfortunately brought up a different problem: Because I used bonferroni as sgRNA integration strategy, plot_response_grna_target_pair gave me error "gRNA target is not present" due to the grna_group_idxs within sceptre_object contains gRNA IDs as singletons. Would you mind explain (1) how the p-values of gRNA IDs with the same targets were combined into 1 single p-value with bonferroni? and (2) How should I then plot this accordingly (e.g., was the gRNA ID with the lowest corrected p-value the final result) ?

**Comment by timothy-barry** [comment]
Hi,

You can read about the `bonferroni` gRNA integration strategy [here](https://timothy-barry.github.io/sceptre-book/methods.html#integrating-information-across-grnas). We need to improve the interface to `plot_response_grna_target_pair()` when `grna_integration_strategy` is `bonferroni`, but yes, my recommendation would be to select the gRNA with the lowest p-value for the plot.
